src/test/ui/inline-const/const-match-pat-range.rs

```diff
@@ -0,0 +1,23 @@
+// build-pass
+
+#![allow(incomplete_features)]
+#![feature(inline_const)]
+fn main() {
+    const N: u32 = 10;
+    let x: u32 = 3;
+
+    match x {
+        const { N - 1 } ..= 10 => {},
+        _ => {},
+    }
+
+    match x {
+        const { N - 1 } ..= const { N + 1 } => {},
+        _ => {},
+    }
+
+    match x {
+        1 ..= const { N - 1 } => {},
+        _ => {},
+    }
```

Comment:
Could you add test cases for half-open ranges as well - `const { N } ..` and `.. const { N }`?

Reply:
Could you add `..= const { .. }` to test cases as well, so all range code paths are covered?

Reply:
Let me know if the added tests are the ones you wanted to include